docs/aws/audit/sgaudit/rules/subnet_exists_without_flow_logs.mdx

```diff
@@ -23,6 +23,274 @@ HIPAA
+
+4. Choose the 'Flow Logs' tab.
+
+5. If there are no flow logs enabled for the selected subnet, it means that the Flow Logs are not enabled. If there are flow logs present, check the 'Status' column. If the status is 'ACTIVE', it means that the Flow Logs are enabled. If the status is 'ERROR' or 'CREATING', it means that there is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can use the following command to list all the VPCs in your account:
+
+   ```
+   aws ec2 describe-vpcs
+   ```
+   This command will return a JSON object with details about all your VPCs.
+
+3. To check if flow logs are enabled on a specific subnet, you need to know the ID of the subnet. You can get this information by running the following command:
+
+   ```
+   aws ec2 describe-subnets
+   ```
+   This command will return a JSON object with details about all your subnets. Look for the "SubnetId" field in the output.
+
+4. Once you have the subnet ID, you can check if flow logs are enabled for that subnet by running the following command:
+
+   ```
+   aws ec2 describe-flow-logs --filter "Name=resource-id,Values=<SubnetId>"
+   ```
+   Replace `<SubnetId>` with the ID of the subnet you are interested in. If flow logs are enabled, this command will return a JSON object with details about the flow logs. If no flow logs are enabled, the command will return an empty JSON object.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install and configure Boto3 to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to allow Boto3 to interact with AWS services on your behalf. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID='your_access_key'
+   AWS_SECRET_ACCESS_KEY='your_secret_key'
+   ```
+
+2. Use Boto3 to interact with AWS EC2 service:
+   You can use Boto3 to interact with AWS EC2 service and retrieve information about your VPCs and subnets. Here is a sample script that lists all the subnets in your account:
+   ```python
+   import boto3
+
+   ec2 = boto3.resource('ec2')
+
+   for subnet in ec2.subnets.all():
+       print(subnet.id)
+   ```
+
+3. Check if flow logs are enabled for each subnet:
+   For each subnet, you can check if flow logs are enabled by calling the `describe_flow_logs` method of the EC2 client and passing the subnet ID as a filter. Here is a sample script that checks if flow logs are enabled for each subnet:
+   ```python
+   import boto3
+
+   ec2 = boto3.client('ec2')
+
+   for subnet in boto3.resource('ec2').subnets.all():
+       response = ec2.describe_flow_logs(
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>